test/ref_dev_examples.cpp

```diff
@@ -0,0 +1,156 @@
+#include <iostream>
+#include <migraphx/program.hpp>
+#include <migraphx/literal.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/verify.hpp>
+#include <migraphx/ref/target.hpp>
+#include <migraphx/onnx.hpp>
+#include "test.hpp"
+
+/*!
+ * Example MIGraphX programs for following the Contributor's Guide.
+ */
+
+void add_two_literals()
```

Comment:
We might not need to do this now, but it would be good to wrap this around the `TEST_CASE()` macro so that the user can pass in specific functions rather than running everything.